src/algorithms/all_paths.c

```diff
@@ -48,6 +48,7 @@ static void _addNeighbors(AllPathsCtx *ctx, LevelConnection *frontier, uint32_t
 		for(uint32_t i = 0; i < neighborsCount; i++) {
 			Edge e = ctx->neighbors[i];
 
+			ASSERT(ctx->edge_idx < Record_length(ctx->r));
```

Comment:
Although a debug assertion, I believe this validation is performed too many times and should be relocated, probably to `AllPathsCtx_New`